tests/test.rs

```diff
@@ -43,9 +43,13 @@ fn test_c_return() {
         200_u8,
         ffi::c_return_unique_ptr_vector_u8().into_iter().sum(),
     );
-    assert_eq!(
-        200.5_f64,
-        ffi::c_return_unique_ptr_vector_f64().into_iter().sum(),
+    assert!(
+        (200.5_f64
+            - ffi::c_return_unique_ptr_vector_f64()
+                .into_iter()
+                .sum::<f64>())
+        .abs()
+            <= f64::EPSILON
```

Comment:
I really don't think it's the best way to write it. For reason that I don't understand, it's not possible to write

```rust
#[allow(clippy::float_cmp)]
assert_eq!(
    200.5_f64,
    ffi::c_return_unique_ptr_vector_f64().into_iter().sum(),
```
I took a look at the implementation of `c_return_unique_ptr_vector_f64`, and I not 100% confident that all numbers can be represented with an exact f64, which would lead to an exact computation:

```
std::unique_ptr<std::vector<double>> c_return_unique_ptr_vector_f64() {
  auto vec = std::unique_ptr<std::vector<double>>(new std::vector<double>());
  vec->push_back(86.0);
  vec->push_back(75.0);
  vec->push_back(30.0);
  vec->push_back(9.5);
  return vec;
}
```

Reply:
This computation is exact in standard IEEE floating point.

Reply:
I tried to print the hexadecimal representation of the number, but without success. The trait isn't implemented for `f64`. That's good to know.

Reply:
It's just integer_mantissa \* 2\^exponent. e.g. 86\*2\^0 + 75\*2\^0 + 30\*2\^0 + 19\*2\^-1 = 401\*2\^-1. You get loss of precision for things like 0.1 which is approximated by 3602879701896397\*2\^-55.

Reply:
That's right, I thought that the issue was between '1.0 + 2.0 != 3.0`, but it's between `0.1 + 0.2 != 0.3`.